File: app/src/main/java/cz/ackee/sample/login/MainActivity.java
package cz.ackee.sample.login;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import cz.ackee.sample.detail.DetailActivity;
import cz.ackee.sample.R;


public class MainActivity extends AppCompatActivity implements ILoginView {


    private LoginPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        presenter = new LoginPresenter();

        Button btn = (Button) findViewById(R.id.btn_login);
        EditText editName = (EditText) findViewById(R.id.edit_email);
        EditText editPass = (EditText) findViewById(R.id.edit_pass);
        btn.setOnClickListener(v -> presenter.login(editName.getText().toString(), editPass.getText().toString()));
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.onViewAttached(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        presenter.onViewDetached();
    }

    @Override
    public void showError() {

    }

    @Override
    public void openDetail() {
        startActivity(new Intent(this, DetailActivity.class));
    }
}

File: app/src/main/java/cz/ackee/sample/login/MainActivity.kt
package cz.ackee.sample.login

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import cz.ackee.sample.R
import cz.ackee.sample.detail.DetailActivity

class MainActivity : AppCompatActivity(), ILoginView {

    private var presenter: LoginPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        presenter = LoginPresenter()

        val btn = findViewById<Button>(R.id.btn_login)
        val editName = findViewById<EditText>(R.id.edit_email)
        val editPass = findViewById<EditText>(R.id.edit_pass)
        btn.setOnClickListener { presenter!!.login(editName.text.toString(), editPass.text.toString()) }
    }

    override fun onResume() {
        super.onResume()
        presenter!!.onViewAttached(this)
    }

    override fun onPause() {
        super.onPause()
        presenter!!.onViewDetached()
    }

    override fun showError() {
    }

    override fun openDetail() {
        startActivity(Intent(this, DetailActivity::class.java))
    }
}
